### DISCUSSION TITLE
Source agreements not being automatically accepted

### DISCUSSION THREAD

**Discussion by dbennett2843** [Romanitho/Winget-AutoUpdate#716] [Q&A]
I've got Sonicwall Netextender installed on a few machines. Winget AutoUpdate can't automatically update the specific build, so I need to uninstall the app and reinstall it.

I created a simple script that calls two lines, the first to uninstall, the second to install the latest version:
   C:\ProgramData\Winget-AutoUpdate\winget-install.ps1 -AppIDs SonicWALL.NetExtender -Uninstall
   C:\ProgramData\Winget-AutoUpdate\winget-install.ps1 -AppIDs SonicWALL.NetExtender -AllowUpgrade
   
When the uninstall runs, it fails to uninstall. I see the following error in the log
The `msstore` source requires that you view the following agreements before using.
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
The source requires the current machine's 2-letter geographic region to be sent to the backend service to function properly (ex. "US").

The command line that WAU is sending includes the "--accept-source-agreements" so I'm puzzled why this isn't working. I'm attaching the install.log.

Any ideas?
[install.log](https://github.com/user-attachments/files/17138469/install.log)

**Comment by KnifMelti** [comment]
It looks like it hs been originally installed from ms store, not from the winget repo.
WAU can't be used for ms store apps. As SYSTEM the store is a no-go. 
Uninstall the ms store app manually and then install it from winget repo. After that WAU can upgrade it.

**Reply by KnifMelti** [reply]
Or just blacklist iit in WAU, it will be automatically upgraded by ms store

**Reply by dbennett2843** [reply]
Interesting...  I pretty confident the install didn't happen via the msstore, but was an install downloaded from the Sonicwall site. Whether that installer hooked to the msstore, I dunno.

But at least that makes sense and I'll know in future if I see msstore appear in the logs, that's probably going to hang us up and we'll need to manually deal with it.

Thanks!

**Reply by KnifMelti** [reply]
Maybe I'm completely wrong (can't check now)...
...the ms store thing in the log may not be important.
It says when trying to upgrade that it's another format. The winget repo is a msi your installation maybe is an exe? :
https://github.com/microsoft/winget-pkgs/blob/master/manifests/s/SonicWALL/NetExtender/10.2.341/SonicWALL.NetExtender.installer.yaml

**Reply by KnifMelti** [reply]
That's probably the culprit here.
Found on SonicWALL that they have both exe and msi while winget repo only have msi.
The msstore part in the log probably comes from the uninstall command lacking `-s, --source`, making winget connect to that repo too when checking. 
Should be addressed in a fix.

**Comment by KnifMelti** [comment]
Solution:
- Installed **NetExtender** via their `NXSetupU-x64-10.2.339.exe` for testing an upgrade
- Created `SonicWALL.NetExtender-preinstall.ps1` (copy `mods\_AppID-template.ps1` to `mods\SonicWALL.NetExtender-preinstall.ps1`)
- Edited `SonicWALL.NetExtender-preinstall.ps1`:
```
#Beginning of App Name string to Silently Uninstall (MSI/NSIS/INNO/EXE with defined silent uninstall in registry)
#Multiple: "app1*","app2*", required wildcard (*) after; search is done with "-like"!
$App = @("SonicWall NetExtender")
```
The result is shown here (after that you can delete `SonicWALL.NetExtender-preinstall.ps1` - WAU will upgrade it correctly the following times):
```
01:41:18 - Checking application updates on Winget Repository...
-> Available update : SonicWall NetExtender. Current version : 10.2.339. Available version : 10.2.341.
01:41:19 - Updating SonicWall NetExtender from 10.2.339 to 10.2.341...
01:41:23 - ##########   WINGET UPGRADE PROCESS STARTS FOR APPLICATION ID 'SonicWALL.NetExtender'   ##########
01:41:23 - Modifications for SonicWALL.NetExtender before upgrade are being applied...
01:41:26 - -> Running: Winget upgrade --id SonicWALL.NetExtender -e --accept-package-agreements --accept-source-agreements -s winget -h


  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 2.76 MB
  █████████████████████▒▒▒▒▒▒▒▒▒  2.00 MB / 2.76 MB
  ██████████████████████████████  2.76 MB / 2.76 MB
No installed package found matching input criteria.
01:41:27 - -> An upgrade for SonicWall NetExtender failed, now trying an install instead... (1/2)
01:41:27 - -> Running: Winget install --id SonicWALL.NetExtender -e --accept-package-agreements --accept-source-agreements -s winget -h --force


  ██████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 2.76 MB
  █████████████████████▒▒▒▒▒▒▒▒▒  2.00 MB / 2.76 MB
  ██████████████████████████████  2.76 MB / 2.76 MB
Found SonicWall NetExtender [SonicWALL.NetExtender] Version 10.2.341
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://software.sonicwall.com/NetExtender/NetExtender-x64-10.2.341.msi


  ████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.53 MB / 15.5 MB
  ████████████████████████████▒▒  14.5 MB / 15.5 MB
  ██████████████████████████████  15.5 MB / 15.5 MB
Successfully verified installer hash
Starting package install...

Successfully installed
01:41:41 - ##########   WINGET UPGRADE PROCESS FINISHED FOR APPLICATION ID 'SonicWALL.NetExtender'   ##########
01:41:41 - SonicWall NetExtender updated to 10.2.341 !
01:41:45 - 1 apps updated ! No more update.

```
